lesson7.md

```diff
@@ -0,0 +1,254 @@
+<!-- .slide: id="lesson7" -->
+
+# Basic Frontend - Spring 2020
+
+Lesson 7, Tuesday, 2020-03-24
+
+---
+
+<!-- .slide: id="lesson7:arrays" -->
+
+### Refresher: Types
+
+```js
+// Strings
+let name = "Owen";
+
+// Numbers
+let age = 28;
+
+// Booleans
+let isProgrammer = true;
+```
+
+---
+
+### Arrays
+
+Array is a container type that holds multiple values:
+
+```js
+let cars = ["Saab", "Volvo", "BMW"];
+
+// we create an array using []
+let emptyArray = [];
+
+// we put the values we want in these angled brackets
+// separated by commas
+let ages = [19, 33, 25, 40];
+```
+
+---
+
+Array can hold any type of value:
+
+```js
+let prices = [0.99, 1.49];
+```
+
+And any quantity:
+```js
+// I only have one favorite food
+let favoriteFoods = ["Pizza"];
+// An array holding 26 letters of the alphabet
+let alphabet = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l"]; //TODO
+```
+
+---
+
+### Analogy: Bookshelf
+
+A bookshelf represents an array of books
+
+![bookshelf](images/bookshelf.jpg)
+
+```js
+let books = ["Harry Potter", "The Hobbit", "Game of Thrones"];
+```
+
+---
+
+### Accessing elements
+
+We can access elements in the array by number using square brackets "`[]`"
+
+The numbering starts at `0`:
```

Comment:
When talking about accessing elements using numbers, I would use a building analogy. Like the array is a building consisting of apartments and on each floor there is someone living on that floor. The building has ground floor, the first floor, etc. When we say array[0] we are basically asking who is living on the ground floor

Reply:
Yes, I was thinking the same thing! Was planning to mention it verbally but I will also put a note in this slide